sdk/nodejs/iot/zMixins.ts

```diff
@@ -0,0 +1,109 @@
+     */
+    consumerGroup?: ConsumerGroup;
+
+    /**
+     * Set to 'many' in order to enable batching. If omitted or set to 'one', single message passed to function.
+     */
+    cardinality?: pulumi.Input<"many" | "one">;
+}>;
+
+declare module "./ioTHub" {
+    interface IoTHub {
+        /**
+         * Subscribes to events logged to this Event Hub to the handler provided, along
+         * with options to control the behavior of the subscription.
+         */
+        onEvent(
+            name: string, args: EventHubCallback | IoTHubSubscriptionArgs, opts?: pulumi.ComponentResourceOptions): IoTHubEventSubscription;
+    }
+}
+
+IoTHub.prototype.onEvent = function(this: IoTHub, name, args, opts) {
+    const functionArgs = args instanceof Function
+        ? <IoTHubSubscriptionArgs>{ callback: args }
+        : args;
+
+    return new IoTHubEventSubscription(name, this, functionArgs, opts);
+}
+
+export class IoTHubEventSubscription extends appservice.EventSubscription<EventHubContext, string, void> {
+    readonly iotHub: IoTHub;
+
+    constructor(
+        name: string, iotHub: IoTHub,
+        args: IoTHubSubscriptionArgs, opts: pulumi.ComponentResourceOptions = {}) {
+
+        opts = { parent: iotHub, ...opts };
+
+        const { resourceGroupName, location } = appservice.getResourceGroupNameAndLocation(args, iotHub.resourceGroupName);
+
+        // The event hub binding does not store the Event Hubs connection string directly.  Instead, the
+        // connection string is put into the app settings (under whatever key we want). Then, the
+        // .connection property of the binding contains the *name* of that app setting key.
+        const bindingConnectionKey = "BindingConnectionAppSettingsKey";
+
+        const bindings: EventHubBindingDefinition[] = [{
+            name: "eventHub",
+            direction: "in",
+            type: "eventHubTrigger",
+            eventHubName: iotHub.name,
+            consumerGroup: args.consumerGroup ? args.consumerGroup.name : "$Default",
+            cardinality: args.cardinality,
+            connection: bindingConnectionKey,
+        }];
+
+        pulumi.all([iotHub.fallbackRoute, iotHub.routes]).apply(([fallbackRoute, routes]) => {
+            if (fallbackRoute && fallbackRoute.enabled) {
+                return;
+            }
+            if (routes && routes.length > 0) {
+                return;
+            }
+            throw new pulumi.ResourceError("IoT Hub must have a route or fallback route enabled.", opts.parent);
+        });
+
+        // Place the mapping from the well known key name to the Event Hubs account connection string in
+        // the 'app settings' object.
+        // The connection string is built from the IoT Hub "event hub compatible endpoint" and the iothubowner access policy key - see https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-messages-read-builtin
```

Comment:
Nit: break the line, it's a bit too long

Reply:
fixed in 83530b3ffd98f0ba8569646e9e665143dd0c827a